File: lib/biz/spec-iot/src/main/java/site/addzero/biz/spec/iot/spi/TelemetryTableNamingStrategy.java
package site.addzero.biz.spec.iot.spi;

import site.addzero.biz.spec.iot.IotThingRef;

/**
 * TDengine table naming strategy.
 */
public interface TelemetryTableNamingStrategy {

    String getName();

    boolean supports(IotThingRef schemaRef);

    String stableTableName(IotThingRef schemaRef);

    String subTableName(IotThingRef schemaRef, IotThingRef sourceRef);
}

File: lib/biz/spec-iot/src/main/kotlin/site/addzero/biz/spec/iot/spi/TelemetryTableNamingStrategy.kt
package site.addzero.biz.spec.iot.spi

import site.addzero.biz.spec.iot.IotThingRef

/**
 * TDengine table naming strategy.
 */
interface TelemetryTableNamingStrategy {

    val name: String

    fun supports(schemaRef: IotThingRef): Boolean

    fun stableTableName(schemaRef: IotThingRef): String

    fun subTableName(schemaRef: IotThingRef, sourceRef: IotThingRef): String
}
